tests/test_init.py

```diff
@@ -36,3 +37,15 @@ def test_large_dependencies_not_imported(module: str):
     https://github.com/pyvista/pyvista/pull/7023
     """
     assert not _module_is_loaded(module), error_msg
+
+
+def test_pyvista_oo_flag():
+    """Test that PyVista works correctly with the -OO optimization flag."""
+    code = """
+import pyvista as pv
+mesh = pv.Sphere()
+""".strip()
```

Comment:
This code executes just fine using `-OO` _without_ this PR. The issue was identified with charts, so let's test that instead. This code _does_ currently fail:
```suggestion
    code = "from pyvista import Charts2D"
```

Reply:
"import pyvista" works because there's a lazy importer that only imports modules when requested.
So a better test would probable be:

``` suggestion
    # Bypass lazy importer and directly import everything from core and plotting
    code = 'from pyvista.core import *; from pyvista.plotting import *'
```